bundles/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/ModbusBitUtilities.java

```diff
@@ -173,7 +183,12 @@ public static DecimalType extractStateFromRegisters(ModbusRegisterArray register
                 ByteBuffer buff = ByteBuffer.allocate(4);
                 buff.put(registers.getRegister(index + 1).getBytes());
                 buff.put(registers.getRegister(index).getBytes());
-                return new DecimalType(buff.order(ByteOrder.BIG_ENDIAN).getFloat(0));
+                try {
+                    return new DecimalType(buff.order(ByteOrder.BIG_ENDIAN).getFloat(0));
```

Comment:
If there are more State types I could understand this API change. But are you sure that an `Optional` or a null value are not expressive enough?

Reply:
I like what you are proposing, makes it more apparent what is returned.

Will change it to use optional

Reply:
Resolved in b20aab0